datafusion/sqllogictest/test_files/timestamps.slt

```diff
@@ -394,12 +503,12 @@ SELECT COUNT(*) FROM ts_data_secs where ts > to_timestamp_seconds('2020-09-08 12
 query PPP
 SELECT to_timestamp(1.1) as c1, cast(1.1 as timestamp) as c2, 1.1::timestamp as c3;
 ----
-1970-01-01T00:00:01.100 1970-01-01T00:00:01.100 1970-01-01T00:00:01.100
+1970-01-01T00:00:01.100 1970-01-01T00:00:00.000000001 1970-01-01T00:00:00.000000001
```

Comment:
Do I read this difference right as now `cast(float_col AS timestamp)` is treated as though it is 1.1 ns where as before it was treated as `1.1` sec?

What do we think about simply not supporting explicit conversion from float --> timestamp to follow the duckdb/postgres model.  That feels far more defensible to me than this behavior which is both different than it was previously *AND* not consistent with other engines

Reply:
Yes, thank you, I double checked and reminded myself of what was going on:

```
DataFusion CLI v48.0.0
> select cast(1.1 as timestamp);
+-------------------------+
| Float64(1.1)            |
+-------------------------+
| 1970-01-01T00:00:01.100 |
+-------------------------+
1 row(s) fetched.
Elapsed 0.005 seconds.

> select cast(column1 as timestamp) from values (1.1);
+-------------------------------+
| column1                       |
+-------------------------------+
| 1970-01-01T00:00:00.000000001 |
+-------------------------------+
1 row(s) fetched.
Elapsed 0.001 seconds.
```

Reply:
Indeed. I tried to capture this in the linked issue https://github.com/apache/datafusion/issues/16636
Thanks for bring this example here.